File: cinebook-frontend/src/services/ticketService.js
import axios from "axios";
import axiosInstance from "./axiosinstance";

export const getMyTickets=()=>{
    return axiosInstance.get("/api/tickets/my-tickets");
};

export const getTicketById=(id)=>{
    return axiosInstance.get(`/api/tickets/${id}`);
}

export const allTickets=()=>{
    return axiosInstance.get("/api/tickets/all");
}
export const getTicketByTicketId=(ticketNumber)=>{
    return axiosInstance.get(`/api/tickets/number/${ticketNumber}`);
}

export const getTicketByBookingId=(bookingId)=>{
    return axiosInstance.get(`/api/tickets/booking/${bookingId}`);
}
export const getActivateTickets=()=>{
    return axiosInstance.get("/api/tickets/ActiveTickets");
}
export const getTicketsByMovieId=(movieId)=>{
    return axiosInstance.get(`/api/tickets/movie/${movieId}`);
}

export const getTicketsBytheatreid=(theatreId)=>{
    return axiosInstance.get(`/api/tickets/theatre/${theatreId}`);



}
export const getTicketByShowDate=(showDate)=>{
    return axiosInstance.get(`/api/tickets/showDate/${showDate}`);
}

File: cinebook-backend/cinebook-backend/src/main/java/com/cinebook/cinebook/service/TicketService.java
package com.cinebook.cinebook.service;

import com.cinebook.cinebook.dto.response.TicketResponseDto;
import com.cinebook.cinebook.repository.TicketRepository;
import org.springframework.data.domain.Page;

import java.time.LocalDate;
import java.util.List;

public interface TicketService {
    TicketResponseDto getTicketById(Long id);
    TicketResponseDto getTicketByticketNumber(String ticketNumber);
    TicketResponseDto getTicketByBooking(Long bookingId);
    List<TicketResponseDto> getActiveTickets();
    Page<TicketResponseDto>getAllTickets(int page, int size, String sortBy,String sortDir);

    //search tickets
    //by movieId
    List<TicketResponseDto>getTicketsByMovie(Long movieId);
    //by theatre id
    List<TicketResponseDto>getTicketsByTheatre(Long theatreId);
    //by show date
    List<TicketResponseDto>getTicketsByshowDate(LocalDate showDate);
    List<TicketResponseDto>getMyTickets();


    List<TicketResponseDto>allTickets();
}
